config/configgrpc/configgrpc.go

```diff
@@ -311,3 +312,10 @@ func GetGRPCCompressionKey(compressionType string) string {
 	}
 	return CompressionUnsupported
 }
+
+var _ configauth.WantsAuthenticators = (*GRPCServerSettings)(nil)
+
+func (a *GRPCServerSettings) InjectAuthenticators(map[configmodels.NamedEntity]configauth.Authenticator) error {
+	fmt.Println("injected into the server settings!")
```

Comment:
This is how I confirmed that the approach works :-) This will be changed/removed once the idea is validated.

Reply:
To clarify if I understand this correctly: here we will store the reference to authenticators somewhere in a GRPCServerSettings field and then will use that field in GRPCServerSettings.ToServerOption to lookup the authenticator by the name specified in the config option and then build the server options, right?
It would be good to show this part too, since this all does not fit in my head fully and I am guessing at this point.

Reply:
I think I made this example more confusing than it should actually be. The final solution will probably have only the Authentication config object as WantsAuthenticators, and it will have a property where users would specify which authenticator to use for this config. The Authentication config is embedded into GRPCServerSettings. The config would be something like this:

```yaml
extensions:
  oidc/tenant1:
    issuer_url: https://tenant1.example.com/
    issuer_ca_path: /etc/pki/tls/cert.pem
    client_id: my-oidc-client
    username_claim: email

  oidc/tenant2:
    issuer_url: https://tenant2.example.com/
    issuer_ca_path: /etc/pki/tls/cert.pem
    client_id: my-oidc-client
    username_claim: email

receivers:
  otlp:
    protocols:
      grpc:
        authentication:
          attribute: authorization
          authenticator: oidc/tenant1

processors:

exporters:
  logging:
    logLevel: debug

service:
  pipelines:
    traces:
      receivers: [otlp]
      processors: []
      exporters: [logging]
```

Reply:
OK, this is useful. I was imagining a different config, but I think what you have is better since it allows to reuse the auth extension in many receivers.